src/node_worker.cc

```diff
@@ -242,6 +245,10 @@ void Worker::Run() {
   {
     Locker locker(isolate_);
     Isolate::Scope isolate_scope(isolate_);
+    // V8 computes its stack limit every time a `Locker` is used based on
+    // --stack-size. Reset it to the correct value.
+    isolate_->SetStackLimit(stack_base_);
```

Comment:
Is this one necessary? If I read V8's source code right, the isolate's StackGuard computes the stack limit from `FLAG_stack_size` when it's first created. Locker's constructor and destructor are then responsible for ensuring that the StackGuard is restored and saved.

If the answer is "yes, it's necessary" then you should probably report this upstream because that sounds like a bug.

Reply:
It does not appear to be necessary, no. I’ve removed it.